lib/web/ui/integration.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ui
+
+import (
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+// IntegrationAWSOIDCSpec contain the specific fields for the `aws-oidc` subkind integration.
+type IntegrationAWSOIDCSpec struct {
+	// RoleARN is the role associated with the integration when SubKind is `aws-oidc`
+	RoleARN string `json:"roleArn,omitempty"`
+}
+
+// Integration describes the base Integration fields (can be read or wri) Integration
+type Integration struct {
+	// Name is the Integration name.
+	Name string `json:"name,omitempty"`
+	// SubKind is the Integration SubKind.
+	SubKind string `json:"subKind,omitempty"`
+	// AWSOIDC contains the fields for `aws-oidc` subkind integration.
+	AWSOIDC *IntegrationAWSOIDCSpec `json:"awsoidc,omitempty"`
```

Comment:
Didn't you use `aws_oidc` in other JSON?

Reply:
This is being used for the HTTP API
I think Typescript, by default, expects the `awsoidc` json tag (this gets treated as a single word)
@kimlisa usually asks for those APIs to use snake_case

If that's not the case, please let me know and I'll change it :+1:

Reply:
it can be anything really, it should really be `awsOidc` but i also thought `awsoidc` was okay too so i didn't say anything. 

and this object is specifically defined for use with web responses (it's defined under `lib/web/ui/*`). ideally all json fields defined in `web/*` should be using `camelCase` but this got lost with new devs defaulting to `snake_case` (which is understandable since everywhere else is that way) and we also started with `snake_case` at the very beginning too so you'll find very old json field names starting with `snake_case`

Reply:
Yes I know it _can_ be anything. My point is just that we used `aws_oidc` here: https://github.com/gravitational/teleport/pull/24145/files#diff-3a5d1089cc78eb7510939ac3850e0e5c54b5d71aea2f0c1a15bcbdc44b94f733R219

I don't care either way, but it feels like using different naming for a similar field in different parts of the codebase is asking for trouble. Can we just pick one and use it consistently?

Reply:
Edit: I see one is the backend resource, where we use snake case, and the other is for the web API where we use difference conventions.